CMakeLists.txt

```diff
@@ -20,7 +20,8 @@ message(STATUS "CMAKE_CXX_COMPILER_VERSION is ${CMAKE_CXX_COMPILER_VERSION}")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
 if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "5.4")
   # we officially support these compiler flags for compiler version 5.4 or less
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-variable -Werror")
+  # We suppress sign-compare error messages because caffe2 headers have these errors.
```

Comment:
Why would we make something non-caffe2 specific if only caffe2 has the problem?

(I'm sure TC also has sign-compare warnings, but then the comment does not make sense).

Reply:
thanks. the comment means that WITH_CAFFE2=ON is needed atm so that sign-compare errors are ignored. this is because the compiler flag is added for it only when caffe2 is build and hence Wno-sign-error is caffe2 build specific flag only.

Reply:
If _only_ caffe2 has these problems, there is no need to make the flag independent of `WITH_CAFFE2`.  If other, non-caffe2 files have these problems, the comment above the line should not say "because caffe2 headers have these errors."

Reply:
okay, thanks. I understand now what you meant to say. I'll remove the comment (it was moved from caffe2 specific section to here)

Reply:
thanks!